Libraries/PeriphDrivers/Include/MAX32657/i3c.h

```diff
@@ -0,0 +1,621 @@
+ * @param   i3c         Pointer to I3C registers.
+ *
+ * @return  The I3C push-pull frequency in hertz.
+ */
+unsigned int MXC_I3C_GetPPFrequency(mxc_i3c_regs_t *i3c);
+
+/**
+ * @brief   Sets the SCL frequency for I3C open-drain operation.
+ *
+ * Note that open-drain SCL also depends on push-pull SCL settings. See
+ * MXC_I3C_SetPPFrequency().
+ *
+ * @param   i3c           Pointer to I3C registers (selects the I3C block used).
+ * @param   frequency     Frequency in hertz.
+ * @param   highPP        Set SCL high period to high period in push-pull mode.
+ *                        This is used to prevent legacy I2C devices from detecting
+ *                        I3C messages.
+ * @return  Negative if error, otherwise actual speed set. See \ref
+ *          MXC_Error_Codes for the list of error return codes.
+ */
+int MXC_I3C_SetODFrequency(mxc_i3c_regs_t *i3c, unsigned int frequency, bool highPP);
+
+/**
+ * @brief   Get the frequency of the I3C open-drain mode.
+ *
+ * @param   i3c         Pointer to I3C registers.
+ *
+ * @return  The I3C open-drain frequency in hertz.
+ */
+unsigned int MXC_I3C_GetODFrequency(mxc_i3c_regs_t *i3c);
+
+/**
+ * @brief   Sets the SCL frequency for I2C mode.
+ *
+ * @param   i3c           Pointer to I3C registers (selects the I3C block used).
+ * @param   frequency     Frequency in hertz.
+ *
+ * @return  Negative if error, otherwise actual speed set. See \ref
+ *          MXC_Error_Codes for the list of error return codes.
+ */
+int MXC_I3C_SetI2CFrequency(mxc_i3c_regs_t *i3c, unsigned int frequency);
+
+/**
+ * @brief   Get the frequency of the I3C in I2C mode.
+ *
+ * @param   i3c         Pointer to I3C registers.
+ *
+ * @return  The frequency of I2C mode in hertz.
+ */
+unsigned int MXC_I3C_GetI2CFrequency(mxc_i3c_regs_t *i3c);
+
+/**
+ * @brief   Sets the skew value for I3C push-pull operation.
+ *
+ * Note that this setting requires peripheral clock (fclk) to SCL ratio to be at
+ * least 4. See MXC_I3C_SetPPFrequency().
+ *
+ * @param   i3c           Pointer to I3C registers (selects the I3C block used).
+ * @param   skew          Skew value in units of peripheral clock cycles. Cannot
+ *                        be greater than 7.
+ *
+ * @return  Success/Fail, see \ref MXC_Error_Codes for a list of return codes.
+ */
+int MXC_I3C_SetSkew(mxc_i3c_regs_t *i3c, uint8_t skew);
+
+/**
+ * @brief   Sets the high-keeper implementation for the device.
+ *
+ * See \ref mxc_i3c_high_keeper_t.
+ *
+ * @param   i3c           Pointer to I3C registers (selects the I3C block used).
+ * @param   hkeep         High-keeper option.
+ *
+ * @return  Success/Fail, see \ref MXC_Error_Codes for a list of return codes.
+ */
+int MXC_I3C_SetHighKeeperMode(mxc_i3c_regs_t *i3c, mxc_i3c_high_keeper_t hkeep);
+
+/**
+ * @brief   Emit an I2C or I3C start.
+ *
+ * @param   i3c         Pointer to I3C registers.
+ * @param   i2c         True if I2C, false if I3C.
+ * @param   readWrite   Direction of transfer. 1 if read, 0 if write.
```

Comment:
You could consider an enum for this.  I think this improves writing/reading the drivers, because it helps avoid a step where you have to look up which value corresponds to a read or write.

i.e.

```C
typdef enum {
    MXC_I3C_TRANSFER_TYPE_READ = 0,
    MXC_I3C_TRANSFER_TYPE_WRITE = 1
} mxc_i3c_transfer_type_t;
```

Reply:
Using enums now.